### DISCUSSION TITLE
How to Manual triggered form validation

### DISCUSSION THREAD

**Discussion by caidaol** [twbs/bootstrap#37396] [Q&A]
Button type of submit can trigger, but how to use javascript to triggered validation?

$('#form1').submit() 
it's not work, it will submit what ever validate state.

help me, thanks.

**Comment by nkdas91** [comment]
Hi @caidaol 

Since you are using jQuery, try this:

```javascript
$("#form1").submit(function (event) {
  // Check validity
  if (!event.target.checkValidity()) {
    // If form is invalid, it won't be submitted
    event.preventDefault();
    event.stopPropagation();
  }

  // If form is valid, it will be submitted
});
```

See CodePen example: https://codepen.io/nkdas91/pen/LYrpWjr
